### PR TITLE
Adding photos to the photo book now uploads them to mongo

### PR DESCRIPTION
**Pull request by Nil-Cipher**
State is retained in between sessions.

### CODE REVIEW

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt [hunk 1]
```diff
@@ -83,39 +97,91 @@ fun capitalize(s: String): String {
...
+                    // TODO: handle failed image creation
+                    println(statusResponse.body)
+                    appState.pageState.value = PageStates.HOME
+                }
+            } catch (e: Exception) {
+                // TODO: handle error
+                println(e.message)
+            }
+            appState.photos.add(
+                0, PhotobookPhoto(
```
Comment (Carson-Tang) [inline ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt:178]: what does this 0 do

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt [hunk 2]
```diff
@@ -83,39 +97,91 @@ fun capitalize(s: String): String {
...
 }
 
-@OptIn(ExperimentalPermissionsApi::class)
+// bitmap -> base64 string
+fun encodeImage(image: Bitmap): String {
+    val stream = ByteArrayOutputStream()
+    image.compress(Bitmap.CompressFormat.PNG, 100, stream)
+    val imageByteArray = stream.toByteArray()
+    val str = Base64.encode(imageByteArray)
+    println("decoded image")
```
Comment (Carson-Tang) [inline ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt:106]: can we remove these prints

**Review by Carson-Tang — APPROVED** [review]

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt (hunk 1 above)
Comment (Nil-Cipher) [reply ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt:178]: It puts the photo at the top of the list, thereby visible to the user. If we put it at the end of the list, the user can't see off first glance that the photo was added.

File: ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt (hunk 2 above)
Comment (Nil-Cipher) [reply ZenJourney/app/src/main/java/ca/uwaterloo/cs/PhotobookPage.kt:106]: Yeah. Will do.

**Pull request merged**